### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'sourceName')

### ISSUE THREAD

**Issue by k06a** [NomicFoundation/hardhat#3670]
### Version of Hardhat

2.12.7

### What happened?

```
21 passing (2s)
  1 failing

/Users/k06a/Projects/limit-order-protocol/node_modules/hardhat/internal/hardhat-network/stack-traces/solidity-errors.js:111
    return new SolidityCallSite(sourceReference.sourceName, sourceReference.contract, sourceReference.function !== undefined
                                                ^

TypeError: Cannot read properties of undefined (reading 'sourceName')
(Use `node --trace-uncaught ...` to show where the exception was thrown)
error Command failed with exit code 7.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Minimal reproduction steps

On the following commit https://github.com/1inch/limit-order-protocol/tree/fcf88a27d6a018074139513cff4b092b76887fa3:
```
yarn && yarn test:ci --bail
```

**Comment by k06a** [comment]
How to fetch full trace for the issue?

**Comment by alcuadrado** [comment]
Hey,

This should be fixed by #3415. I published a pre-release version of it as `hardhat@viair` and run your tests and now I get 

>      AssertionError: Expected transaction to be reverted with custom error 'RFQTakingAmountTooHigh', but it reverted with panic code 0x11 (Arithmetic operation underflowed or overflowed outside of an unchecked block)

Can you check if that error is correct?

**Comment by k06a** [comment]
@alcuadrado seems like something which can be correct for sure.

**Comment by panukettu** [comment]
> This should be fixed by #3415. I published a pre-release version of it as `hardhat@viair` 

Resolved the error for me. 

`MultiSolcUserConfig` compiler entry

```
{
    version: "0.8.14",
    settings: {
        viaIR: true,
        optimizer: {
            enabled: true,
            runs: 200,
            details: {
                constantOptimizer: false,
                deduplicate: true,
                yul: true,
            },
        },
        outputSelection: {
            "*": {
                "*": [
                    "metadata",
                    "abi",
                    "storageLayout",
                    "evm.methodIdentifiers",
                    "devdoc",
                    "userdoc",
                    "evm.gasEstimates",
                    "evm.byteCode",
                ],
            },
        },
    },
},
```

**Comment by fvictorio** [comment]
Pretty sure this is caused by #3365, so I'm tentatively closing this.

**Comment by hcheng826** [comment]
> Hey,
> 
> This should be fixed by #3415. I published a pre-release version of it as `hardhat@viair` and run your tests and now I get
> 
> > ```
> >  AssertionError: Expected transaction to be reverted with custom error 'RFQTakingAmountTooHigh', but it reverted with panic code 0x11 (Arithmetic operation underflowed or overflowed outside of an unchecked block)
> > ```
> 
> Can you check if that error is correct?

sorry can you share more about how to apply the fix?
